quixstreams/dataframe/windows/base.py

```diff
@@ -196,6 +183,201 @@ def window_callback(
 
         return self._apply_window(func=window_callback, name=self._name)
 
+    # Implemented by SingleAggregationWindowMixin and MultiAggregationWindowMixin
+    # Single aggregation and multi aggregation windows store aggregations and collections
+    # values in a different format.
+    @property
+    @abstractmethod
+    def collect(self) -> bool: ...
+
+    @property
+    @abstractmethod
+    def aggregate(self) -> bool: ...
+
+    @abstractmethod
+    def _initialize_value(self) -> Any: ...
+
+    @abstractmethod
+    def _aggregate_value(self, state_value: Any, value: Any) -> Any: ...
+
+    @abstractmethod
+    def _collect_value(self, value: Any): ...
+
+    @abstractmethod
+    def _results(
+        self,
+        aggregated: Any,
+        collected: list[Any],
+        start: int,
+        end: int,
+    ) -> WindowResult: ...
+
+
+class SingleAggregationWindowMixin:
+    """
+    DEPRECATED: Use MultiAggregationWindowMixin instead.
+
+    Single aggregation window mixin for windows with a single aggregation or collection.
+    Store aggregated value directly in the window value.
+    """
+
+    def __init__(
+        self,
+        *,
+        aggregators: dict[str, BaseAggregator],
+        collectors: dict[str, BaseCollector],
+        **kwargs,
+    ) -> None:
+        if (len(collectors) + len(aggregators)) > 1:
+            raise ValueError("Only one aggregator or collector can be defined")
+
+        if len(aggregators) > 0:
+            self._aggregator: Optional[BaseAggregator] = aggregators["value"]
+            self._collector: Optional[BaseCollector] = None
+        elif len(collectors) > 0:
+            self._collector = collectors["value"]
+            self._aggregator = None
+        else:
+            raise ValueError("At least one aggregator or collector must be defined")
+
+        super().__init__(**kwargs)
+
+    @property
+    def aggregate(self) -> bool:
+        return self._collector is None
+
+    @property
+    def collect(self) -> bool:
+        return self._aggregator is None
```

Comment:
I think this will be more accurate
```suggestion
    @property
    def aggregate(self) -> bool:
        return self._aggregator is not None

    @property
    def collect(self) -> bool:
        return self_collector is not None
```